data/processing.py

```diff
@@ -221,138 +199,134 @@ def load_data():
         continue
 
       domain = row[0].lower()
-      if not domain_data.get(domain):
+      if not domains.get(domain):
         # print("[analytics] Skipping %s, not a federal domain from domains.csv." % domain)
         continue
 
       # If it didn't appear in the inspect data, skip it, we need this.
-      if not domain_data[domain].get('inspect'):
-        # print("[analytics] Skipping %s, did not appear in inspect.csv." % domain)
-        continue
+      # if not domains[domain].get('inspect'):
+      #   print("[analytics] Skipping %s, did not appear in inspect.csv." % domain)
+      #   continue
 
       dict_row = {}
       for i, cell in enumerate(row):
         dict_row[headers[i]] = cell
 
-      domain_data[domain]['analytics'] = dict_row
+      scan_data[domain]['analytics'] = dict_row
+
+  return scan_data
 
 # Given the domain data loaded in from CSVs, draw conclusions,
 # and filter/transform data into form needed for display.
-def process_domains():
-
-  # First, process all domains.
-  for domain in domains:
-    if evaluating_for_https(domain):
-      row = https_row_for(domain)
-      domain_map[domain]['https'] = row
-      https_domains.append(row)
-
-    if evaluating_for_analytics(domain):
-      row = analytics_row_for(domain)
-      domain_map[domain]['analytics'] = row
-      analytics_domains.append(row)
-
-  # Second, process each agency's domains.
-  for agency in agencies:
-
-    https_total = 0
-    https_stats = {
-      'https': 0,
-      'https_forced': 0,
-      'hsts': 0,
-      'grade': 0
-    }
+def process_domains(domains, agencies, scan_data):
+
+  reports = {
+    'analytics': {},
+    'https': {}
+  }
+
+  # For each domain, determine eligibility and, if eligible,
+  # use the scan data to draw conclusions.
+  for domain_name in domains.keys():
+
+    if eligible_for_analytics(domains[domain_name]):
+      reports['analytics'][domain_name] = analytics_report_for(
+        domain_name, domains[domain_name], scan_data
+      )
+
+    if eligible_for_https(domains[domain_name]):
+      reports['https'][domain_name] = https_report_for(
+        domain_name, domains[domain_name], scan_data
+      )
+
+  return reports
+
+# Go through each report type and add agency totals for each type.
+def update_agency_totals():
+  all_agencies = Agency.all()
+
+  # For each agency, update their report counts for every domain they have.
+  for agency in all_agencies:
+
+    # TODO: Do direct DB queries for answers, rather than iterating.
```

Comment:
Right now to calculate/cache totals per-agency, it goes through each agency and then fetches all eligible domains and counts them up in Python code. This should be completely doable with a single database query per-field instead.